llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -3338,14 +3341,13 @@ class BoUpSLP {
       copy(OpVL, Operands[OpIdx].begin());
     }
 
-    /// Set this bundle's operand from \p VL.
-    void setOperand(ArrayRef<Value *> VL, const BoUpSLP &R,
-                    bool RequireReorder = false) {
-      VLOperands Ops(VL, R);
+    /// Set this bundle's operand from Scalars.
+    void setOperand(const BoUpSLP &R, bool RequireReorder = false) {
+      auto *I0 = cast<Instruction>(*find_if(Scalars, IsaPred<Instruction>));
```

Comment:
You still looking for the instruction here, better to pass it as an argument or make it part of VLOperands class, just like Scalars

Reply:
`I0` is passed to `VLOperands Ops(Scalars, R, I0);` to reduce the search. Is this not good?

Reply:
You still redo search here